src/common/class-syntax.k

```diff
@@ -0,0 +1,125 @@
+syntax ClassDec ::= EnumDec
+
+syntax ClassBody ::= "{" ClassBodyDecList "}" [klabel('ClassBody)]
+//todo optionals
+syntax ClassDecHead ::= AnnoClassModList "class" Id TypeParams Super Interfaces [prefer, klabel('ClassDecHead)]
+
+syntax ClassMod ::= Public
+                  | Private
+                  | Protected
+                  | Abstract
+                  | Final
+                  | Static
+                  | StrictFP
+syntax AnnoClassMod ::= Anno | ClassMod
+
+syntax Super ::= "extends" ClassType                    [klabel('SuperDec)]
+syntax Interfaces ::= "implements" InterfaceTypeList    [klabel('ImplementsDec)]
+
+syntax ClassBodyDec ::= InstanceInit | StaticInit | ClassMemberDec | ConstrDec
+
+syntax ClassMemberDec ::= FieldDec | ClassDec | MethodDec | InterfaceDec | SemiColon
+
+syntax SemiColon ::= ";" [klabel('SemiColon)]
+
+//@ \subsection{ConstructorDeclarations.sdf}
+//todo optionals
+syntax ConstrHead   ::=  AnnoConstrModList TypeParams Type Id "(" Params ")" Throws [prefer, klabel('ConstrDecHead)]
+                       | AnnoConstrModList TypeParams Type Id "(" Params ")"    [klabel('ConstrDecHead)]
+                       | AnnoConstrModList Type Id "(" Params ")" Throws   [klabel('ConstrDecHead)]
+                       | AnnoConstrModList Type Id "(" Params ")"    [klabel('ConstrDecHead)]
+
+syntax ConstrDec ::= ConstrHead ConstrBody                    [klabel('ConstrDec)]
+syntax ConstrBody ::= "{" ConstrInv BlockStmList "}"          [prefer, klabel('ConstrBody)]
+                    | "{" BlockStmList "}"                    [klabel('ConstrBody)]
+
+syntax ConstrInv ::= TypeArgs "this"  "(" Exps ")" ";"        [prefer, klabel('AltConstrInv)]
+syntax ConstrInv ::= "this"  "(" Exps ")" ";"                 [klabel('AltConstrInv)]
+syntax ConstrInv ::= TypeArgs "super"  "(" Exps ")" ";"       [prefer, klabel('SuperConstrInv)]
+syntax ConstrInv ::= "super"  "(" Exps ")" ";"                [klabel('SuperConstrInv)]
+syntax ConstrInv ::= Exp "." TypeArgs "super"  "(" Exps ")" ";" [prefer, klabel('QSuperConstrInv)]
+syntax ConstrInv ::= Exp "." "super"  "(" Exps ")" ";"          [klabel('QSuperConstrInv)]
+
+syntax ConstrMod ::= Public
+                  | Private
+                  | Protected
+syntax AnnoConstrMod ::= Anno | ConstrMod
+
+//@ \subsection{EnumDeclarations.sdf}
+syntax EnumDecHead ::= AnnoClassModList "enum" Id Interfaces [prefer, klabel('EnumDecHead)]
+                     | AnnoClassModList "enum" Id            [klabel('EnumDecHead)]
+
+syntax EnumDec ::= EnumDecHead EnumBody                     [klabel('EnumDec)]
+
+syntax EnumBody ::= "{" EnumConstList EnumBodyDecs"}"       [prefer, klabel('EnumBody)]
+                  | "{" EnumConstList "}"                   [klabel('EnumBody)]
+                  | "{" EnumConstList "," EnumBodyDecs "}"  [prefer, klabel('EnumBody)]
+                  | "{" EnumConstList "," "}"               [klabel('EnumBody)]
```

Comment:
I find this syntax design decision of Java rather strange, but I believe you (and I don't feel like checking).  But please put a comment there saying so then, because otherwise people looking at your definition will think that you made a mistake there, like I did.